conans/client/toolchain/cmake.py

```diff
@@ -0,0 +1,316 @@
+                                          make_program=make_program, parallel=parallel,
+                                          generator=self._generator,
+                                          set_cmake_flags=False,
+                                          output=self._conanfile.output)
+        self.definitions = Definitions()
+        self.definitions.update(builder.get_definitions())
+        # FIXME: Removing too many things. We want to bring the new logic for the toolchain here
+        # so we don't mess with the common code.
+        self.definitions.pop("CMAKE_BUILD_TYPE", None)
+        self.definitions.pop("CONAN_IN_LOCAL_CACHE", None)
+        self.definitions.pop("CMAKE_PREFIX_PATH", None)
+        self.definitions.pop("CMAKE_MODULE_PATH", None)
+        self.definitions.pop("CONAN_LINK_RUNTIME", None)
+        for install in ("PREFIX", "BINDIR", "SBINDIR", "LIBEXECDIR", "LIBDIR", "INCLUDEDIR",
+                        "OLDINCLUDEDIR", "DATAROOTDIR"):
+            self.definitions.pop("CMAKE_INSTALL_%s" % install, None)
+
+    def _deduce_fpic(self):
+        fpic = self._conanfile.options.get_safe("fPIC")
+        if fpic is None:
+            return None
+        os_ = self._conanfile.settings.get_safe("os")
+        if os_ and "Windows" in os_:
+            self._conanfile.output.warn("Toolchain: Ignoring fPIC option defined for Windows")
+            return None
+        shared = self._conanfile.options.get_safe("shared")
+        if shared:
+            self._conanfile.output.warn("Toolchain: Ignoring fPIC option defined "
+                                        "for a shared library")
+            return None
+        return fpic
+
+    def _deduce_vs_static_runtime(self):
+        settings = self._conanfile.settings
+        if (settings.get_safe("compiler") == "Visual Studio" and
+                "MT" in settings.get_safe("compiler.runtime")):
+            return True
+        return False
+
+    def dump(self, install_folder):
+        conan_project_include_cmake = os.path.join(install_folder, "conan_project_include.cmake")
+        conan_project_include_cmake = conan_project_include_cmake.replace("\\", "/")
+        t = Template(self._template_project_include)
+        content = t.render(configuration_types_definitions=self.definitions.configuration_types,
+                           vs_static_runtime=self._vs_static_runtime)
+        save(conan_project_include_cmake, content)
+
+        # TODO: I need the profile_host and profile_build here!
+        # TODO: What if the compiler is a build require?
+        # TODO: Add all the stuff related to settings (ALL settings or just _MY_ settings?)
+        # TODO: I would want to have here the path to the compiler too
+        build_type = self._build_type if not is_multi_configuration(self._generator) else None
+        try:
+            # This is only defined in the cache, not in the local flow
```

Comment:
Is this not for `ConanFile.in_local_cache` is for?

Reply:
The rationale when I coded that was that I didn't want to rely on that, I was aiming to have exactly the same experience both in the local cache and in the user local flow. So as long as the "package_folder" is defined, I would like this code to work. We need to think about this local flow with toolchains, this is a minor implementation detail.

Reply:
On another note, why use `package_folder` instead of `install_folder`?